lib/get-specs-from-issue.js

```diff
@@ -2,38 +2,81 @@ module.exports = getSpecsFromIssue;
-  let specs = [];
+  const repo = context.repo();
 
   return Promise.resolve()
-  .then(() => Promise.all([
-    context.github.issues.get(...repo, issueNumber),
-    context.github.issues.listComments(...repo, issueNumber)
-  ]))
-  .then(data => {
-    const { body, comments } = data;
-    const allCommentsBody = comments
-      .map((comment) => comment.body)
-      .push(body);
-
-    pushSpecs(specs, allCommentsBody);
-
-    return specs;
-  });
+    .then(() => Promise.all([
+      context.github.issues.get(...repo, issueNumber),
+      context.github.issues.listComments(...repo, issueNumber)
+    ]))
+    .then(data => {
+      const { issueBody, issueComments } = data;
+
+      return issueComments
+        .map(comment => comment.body)
+        .push(issueBody)
+        .join('\n');
+    })
+    .then(issueCompleteText => getSpecsArray(issueCompleteText))
+    .then(specsArray => specNamesFromMarkdownList(specsArray));
 }
 
-function pushSpecs(specs, body) {
-  body.toLowerCase();
-  let posiotionOfSpecsList = body.indexOf('## specs');
-  body = body.slice(posiotionOfSpecsList);
-  let endPosiotionOfSpecsList = body.indexOf('##');
-  body = body.slice(0, endPosiotionOfSpecsList);
-
-  let bullets = body.split('-');
-  bullets.forEach(bullet => {
-    bullet.trim();
-    if (bullet.startsWith('it')) {
-      specs.push(bullet);
+function getSpecsArray(completeText) {
+  const specsBlockStartRegex = /^\s*?(#+)\s*?specs/gi
+  const specsArray = [];
+  var isInsideSpecsBlock = false;
+  var specsBlockTopLevelHeader;
+
+  completeText.split('\n').forEach(line => {
+    if (!isInsideSpecsBlock) {
+      const lineMatchResult = specsBlockStartRegex.exec(line);
+      if (lineMatchResult) {
+        isInsideSpecsBlock = true;
+        specsBlockTopLevelHeader = lineMatchResult[1];
+      }
+    } else {
+      const isHigherLevelMarkdownTitle = isMarkdownTitle(line) &&
+        !isMarkdownTitleDepthValid(line, specsBlockTopLevelHeader);
+
+      const isLineValid = isEmptyLine(line) ||
+        isMarkdownListItem(line) ||
+        isMarkdownTitle(line) &&
+          isMarkdownTitleDepthValid(line, specsBlockTopLevelHeader);
+
+      if (isHigherLevelMarkdownTitle) {
+        isInsideSpecsBlock = false;
+      } else if (isLineValid) {
+        specsArray.push(line);
+      }
     }
   });
+
+  return specsArray;
+}
+
+function isMarkdownTitle(line) {
+  return line.startsWith('#');
+}
+
+function isMarkdownTitleDepthValid(title, rootTitlePrefix) {
+  return !title.startsWith(rootTitlePrefix)
+} 
+
+function isMarkdownListItem(line) {
+  const markdownListItemRegex = /^\s*?[-\*](?:.*)/g
+  return markdownListItemRegex.test(line);
+}
+
+function isEmptyLine(line) {
+  return line.trim() === '';
+}
+
+function specNamesFromMarkdownList(specsArray) {
+  const specs = [];
+  specsArray.forEach(spec => {
+    const specName = spec.replace(/[-\*]*/g, '').trim();
+    specs.push(specName);
+  });
+
+  return specs;
```

Comment:
- this can be replaced with a map.
- this is not the correct regex. it will remove all dashes and asteriks in the spec name as well
- I think a regex is an overkill here. you basically just need to remove the first character (since you already trimmed each line), and then trim again to remove any trailing whitespaces. but I don't mind you using a regex, just create one that will clean just the first one. checkout the regex from my previous code review for a regex that checks the first one only. you can use [regex101](https://regex101.com/#javascript) to see an analysis on what the regex does exactly.

```javascript
function specNamesFromMarkdownList(listItems) {
  return listItems.map(listItem => /* do things to the item */);
}
```